src/cc/libbpf.c

```diff
@@ -177,9 +187,14 @@ int bpf_prog_load(enum bpf_prog_type prog_type,
   if (ret < 0 && !log_buf) {
     // caller did not specify log_buf but failure should be printed,
     // so call recursively and print the result to stderr
+    buffer_size = attr.insn_cnt * 128;
```

Comment:
I understand your point. If we have to try few times do you have a suggestion regarding the memory limit? Basically when shall we give up ?
Also if we need to implement this for a c api, who owns the memory in this case?